purchases/src/main/kotlin/com/revenuecat/purchases/listenerConversions.kt

```diff
@@ -128,14 +120,14 @@ fun Purchases.getOfferingsWith(
 fun Purchases.purchaseProductWith(
     activity: Activity,
     storeProduct: StoreProduct,
-    onError: PurchaseErrorFunction = ON_PURCHASE_ERROR_STUB,
-    onSuccess: PurchaseCompletedFunction
+    onError: (error: PurchasesError, userCancelled: Boolean) -> Unit = ON_PURCHASE_ERROR_STUB,
+    onSuccess: (purchase: StoreTransaction, customerInfo: CustomerInfo) -> Unit
```

Comment:
Oh shoot I just realized we have an inconsistency on the order of the functions. Here we have `onError` first then `onSuccess`, which is on purpose so folks can omit the last parameter and do:

```
Purchases.purchaseProductWith(.....) { purchase, customerInfo ->
```

But other functions have the `onError` in last place 😮‍💨 

Nothing to worry about now but wanted to mention it

Reply:
Ohh good catch! Yeah... I guess we might want to make it consistent on the next major... But probably we shouldn't change it here.